src/artifactorypurge/artifactorypurge.py

```diff
@@ -1,64 +1,78 @@
+import os
+
 import click
-import imp
-from .utils.logging import getLogger
-from .credentials import load_credentials
+from pluginbase import PluginBase
 
+from .credentials import load_credentials
 from .utils.artifactory import Artifactory
+from .utils.logging import getLogger
 from .utils.performance import get_performance_report
 
 LOG = getLogger(__name__)
 
+
 @click.command()
-@click.option('--dryrun/--nodryrun', default=True, is_flag=True, help='Dryrun does not delete any artifacts. On by default')
-@click.option('--reponame', default=None, help='Operate on single repo')
-@click.option('--project', default=None, help='Operate on single project')
+@click.option(
+    '--dryrun/--nodryrun', default=True, is_flag=True, help='Dryrun does not delete any artifacts. On by default')
+@click.option('--plugin-path', required=True, help='Path to plugin directory')
```

Comment:
I believe this should be optional. I should just be able to run it.
A normal person may not care about the plugins. What if they don't have any custom ones? What would they enter?

Reply:
agreed. I will update that